src/Search/SearchContext.tsx

```diff
@@ -0,0 +1,27 @@
+import React, {createContext, ReactNode, useContext, useState} from 'react'
+
+interface SearchContextApi {
+  search: string
+  setSearch: (search: string) => void
+  clearSearch: () => void
+  visible: boolean
+  setVisible: (visible: boolean) => void
+}
+
+const SearchContext = createContext<SearchContextApi | null>(null)
+
+export const useSearch = () => {
+  const value = useContext(SearchContext)
+  if (!value) {
+    throw new Error('useSearch must be used within a SearchProvider')
+  }
+  return value
+}
+
+export const SearchProvider = ({children, value}: {children: ReactNode; value?: Partial<SearchContextApi>}) => {
+  const [search, setSearch] = useState('')
+  const [visible, setVisible] = useState(false)
+  const clearSearch = () => setSearch('')
+  const defaultValue = {search, setSearch, clearSearch, visible, setVisible}
```

Comment:
what is default value?

Reply:
is visibility toggled from anywhere except the header?
if not, would it make sense to move visibility out of the provider and into the header?

Reply:
or maybe into the navigator file